apps/designer/app/designer/features/style-panel/controls/text/text-control.tsx

```diff
@@ -0,0 +1,251 @@
+import { ChangeEvent, FocusEvent, KeyboardEvent, useRef } from "react";
+import {
+  Box,
+  Text,
+  Grid,
+  IconButton,
+  TextField,
+  Tooltip,
+  IconButtonWithMenu,
+  numericScrubControl,
+} from "@webstudio-is/design-system";
+import { getFinalValue } from "../../shared/get-final-value";
+import { useIsFromCurrentBreakpoint } from "../../shared/use-is-from-current-breakpoint";
+import { ControlProps } from "../../style-sections";
+import { iconConfigs, StyleConfig } from "../../shared/configs";
+import { type StyleValue, type Unit, units } from "@webstudio-is/react-sdk";
+import { ChevronDownIcon } from "@webstudio-is/icons";
+import { PropertyName } from "../../shared/property-name";
+
+const commonUnitsSubset = units
+  .slice(0)
+  .sort((v) =>
+    ["%", "px", "rem", "em", "ch", "vh", "vw", "hv", "vmin", "vmax"].includes(v)
+      ? -1
+      : 1
+  );
+
+const CSSGridLeft = {
+  "& ~ input": {
+    paddingLeft: "calc($sizes$6 + 6px)",
+  },
+  gridArea: "1 / 1 / 2 / 2",
+};
+
+const CSSGridRight = {
+  gridArea: "-1 / -1 / -2 / -2",
+  "& ~ input": {
+    paddingRight: "calc($sizes$5 + 6px)",
+  },
+};
+
+const CSSIconButton = {
+  zIndex: 1,
+  borderRadius: 2,
+  height: "calc($sizes$6 - 6px)",
+  width: "calc($sizes$6 - 6px)",
+  transform: "translate(3px, 3px)",
+  "&:focus": {
+    boxShadow: "none",
+  },
+};
+
+const CSSTextUnit = {
+  fontSize: "$1",
+  cursor: "default",
+};
+
+const CSSTextField = {
+  height: "$6",
+  fontWeight: "500",
+  paddingLeft: "calc($sizes$1 + 6px)",
+  gridArea: "1 / 1 / -1 / -1",
+  cursor: "default",
+};
+
+export const TextControl = ({
+  category,
+  currentStyle,
+  inheritedStyle,
+  setProperty,
+  styleConfig,
+}: ControlProps) => {
+  const numericScrubRef = useRef<{ disconnectedCallback: () => void }>();
+  const isCurrentBreakpoint = useIsFromCurrentBreakpoint(styleConfig.property);
+
+  const value = getFinalValue({
+    currentStyle,
+    inheritedStyle,
+    property: styleConfig.property,
+  });
+
+  if (value === undefined) return null;
+
+  // @todo setValue has a latency issue when updating a value fast(i.e keyboard arrows/scrub control)
+  const setValue = setProperty(styleConfig.property);
+
+  let Icon;
+
+  const { label, property } = styleConfig;
+
+  // @todo this silo's the flex section
+  let multiColumn = true;
+  if (
+    category === "layout" &&
+    property.includes("Gap") &&
+    (currentStyle.display?.value as string).includes("flex")
```

Comment:
why is this having specific property checks? It shouldn't , its not a text control any more ... instead you need a property that does the same thing you need but semantically is not attached to the specific property, same in cases below